core/dbt/events/types.py

```diff
@@ -393,6 +393,19 @@ def message(self):
         return warning_tag(msg)
 
 
+@dataclass
+class EnvironmentVariableRenamed(WarnLevel, pt.EnvironmentVariableRenamed):  # noqa
+    def code(self):
+        return "D009"
+
+    def message(self):
+        description = (
+            f"The environment variable `{self.old_name}` has been renamed to `{self.new_name}`. "
+            f"Since `{self.old_name}` has been set, its value will be used instead of `{self.new_name}`."
+        )
```

Comment:
This message is lacking. @dbeatty10 Do you have a suggestion for a better message to communicate that
- `ENV_VAR_FOO` has been deprecated in favor of `ENV_VAR_BAR`
- If you set `ENV_VAR_FOO` its will be used instead of the value of `ENV_VAR_BAR`

Reply:
What do you think of this?
```shell
The environment variable `ENV_VAR_FOO` has been deprecated and renamed as `ENV_VAR_BAR`.
If `ENV_VAR_FOO` is currently set, its value will be used instead of `ENV_VAR_BAR`.
Set `ENV_VAR_BAR` and unset `ENV_VAR_FOO` to avoid this deprecation warning and ensure it works properly in a future release.
```

It's kinda wordy, but it communicates three key things:
1. The name you are using is deprecated and it has a new name.
2. Here's how it will behave in the meantime.
3. Here's the action you should take.

Reply:
That sounds great!